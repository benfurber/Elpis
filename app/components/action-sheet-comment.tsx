import React, { Component } from "react";
import ActionSheet from "react-native-actionsheet";

import { Comment, NavigationType } from "interfaces";
import { labels } from "labels";

interface Props {
  item: Comment;
  navigation: NavigationType;
  refProp: any;
}

class ActionSheetComment extends Component<Props> {
  onPress(index) {
    if (index === 1) console.log("Delete");
    if (index === 2) console.log("Update");
  }

  render() {
    const { refProp } = this.props;
    const { cancel } = labels;

    const options = [cancel, "Delete comment", "Edit comment"];

    return (
      <ActionSheet
        ref={refProp}
        options={options}
        cancelButtonIndex={0}
        onPress={index => this.onPress(index)}
      />
    );
  }
}

export { ActionSheetComment };
