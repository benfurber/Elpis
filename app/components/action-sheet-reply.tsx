import React, { Component } from "react";
import ActionSheet from "react-native-actionsheet";

import { NavigationType, Reply } from "interfaces";
import { labels } from "labels";
import { DELETE_REPLY } from "mutations";
import { client } from "utils";

interface Props {
  item: Reply;
  navigation: NavigationType;
  refProp: any;
}

class ActionSheetReply extends Component<Props> {
  async actionSheetOnPress(index) {
    if (index === 1) {
      const { id } = this.props.item;
      this.setState({ deleting: true });

      await client.mutate({
        mutation: DELETE_REPLY,
        variables: {
          id,
        },
      });
    }

    if (index === 2) {
      const { item, navigation } = this.props;
      const { comment, content, id, imagePath } = item;

      const commentId = comment.id;
      const currentReply = { content, id, imagePath };

      navigation.navigate("AddReply", { commentId, currentReply });
    }
  }

  render() {
    const { refProp } = this.props;
    const { cancel, deleteYourReply, editReply } = labels;

    return (
      <ActionSheet
        ref={refProp}
        options={[cancel, deleteYourReply, editReply]}
        cancelButtonIndex={0}
        onPress={index => this.actionSheetOnPress(index)}
      />
    );
  }
}

export { ActionSheetReply };
