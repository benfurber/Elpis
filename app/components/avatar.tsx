import React, { Component } from "react";
import { Image, View } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";

import { Author, NavigationType } from "interfaces";
import { elements } from "styles";

const annonProfilePath = "../assets/images/profile-pic-annon.png";

interface Props {
  avatarPath: null | Author["avatarPath"];
  containerStyles?: object;
  navigation?: NavigationType;
  userId?: Author["id"];
  size?: "small" | "medium" | "large" | "xl" | "feature";
  styles?: object;
}

class Avatar extends Component<Props> {
  onPress() {
    const { navigation, userId } = this.props;

    if (navigation && userId) {
      navigation.navigate("UserProfile", {
        navigation,
        userId,
      });
    }
  }

  render() {
    const { avatarPath, containerStyles, size, styles } = this.props;

    const path = avatarPath ? { uri: avatarPath } : require(annonProfilePath);

    const styleSet = {
      feature: elements.imageRoundFeature,
      large: elements.imageRoundLarge,
      medium: elements.imageRound,
      small: elements.imageRoundSmall,
      xl: elements.imageRoundXL,
    };

    return (
      <View style={containerStyles}>
        <TouchableOpacity onPress={() => this.onPress()}>
          <Image source={path} style={[styleSet[size || "medium"], styles]} />
        </TouchableOpacity>
      </View>
    );
  }
}

export { Avatar };
