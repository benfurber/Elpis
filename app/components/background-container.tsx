import React from "react";
import { ImageBackground, SafeAreaView, StyleSheet, View } from "react-native";

import { colours } from "styles";

interface Props {
  children: object;
  style?: object;
  viewStyle?: object;
}

const BackgroundContainer = (props: Props) => {
  return (
    <ImageBackground
      source={require("../assets/images/background.png")}
      style={props.style || styles.container}
    >
      <SafeAreaView style={styles.safeArea}>
        <View style={props.viewStyle || styles.viewContainer}>
          {props.children}
        </View>
      </SafeAreaView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colours.transparentBlue,
    height: "100%",
    width: "100%",
  },
  safeArea: {
    flex: 1,
  },
  viewContainer: {
    flex: 1,
    flexDirection: "column",
  },
});

export { BackgroundContainer };
