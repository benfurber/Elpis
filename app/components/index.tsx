export { Comments } from "./comments";
export { BackgroundContainer } from "./background-container";
export { BackgroundModal } from "./background-modal";
export { Badge } from "./badge";
export { ButtonSubmit } from "./button-submit";
export { FormAddProfilePicture } from "./form-add-profile-picture";
export { FormAddPassword } from "./form-add-password";
export { Icon } from "./icon";
export { LoginForm } from "./login-form";
export { Logo } from "./logo";
export { MessageBox } from "./message-box";
export { Post } from "./post";
export { ProfilePictureField } from "./profile-picture-field";
export { TextField } from "./text-field";
export { TextInput } from "./text-input";
export { Title } from "./title";
