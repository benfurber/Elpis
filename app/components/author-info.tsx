import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";

import { Avatar, Title } from "components";
import { Comment, NavigationType } from "interfaces";
import { colours, elements, layout } from "styles";
import { formatDate } from "utils";

interface Props {
  item: Comment;
  navigation: NavigationType;
}

class AuthorInfo extends Component<Props> {
  render() {
    const { item, navigation } = this.props;
    const { author, publishedAt, title } = item;

    return (
      <View style={styles.container}>
        <View style={styles.avatarContainer}>
          <Avatar
            avatarPath={author.avatarPath}
            navigation={navigation}
            userId={author.id}
            size="large"
          />
        </View>
        <View style={styles.titleContainer}>
          {title && <Title text={title} />}
          <Text style={styles.metaDetails}>
            <Text style={styles.name}>{`${author.name} `}</Text>
            {formatDate(publishedAt, false)}
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  avatarContainer: {
    alignItems: "center",
    flexDirection: "row",
    width: 70,
  },
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  metaDetails: {
    marginBottom: layout.spacing,
    ...elements.textDate,
  },
  name: {
    color: colours.navyBlueDark,
    fontWeight: "bold",
  },
  text: {
    flex: 1,
    flexWrap: "nowrap",
    marginRight: layout.spacingXL,
  },
  titleContainer: {
    flex: 1,
  },
});

export { AuthorInfo };
