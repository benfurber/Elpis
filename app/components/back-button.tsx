import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import { Icon } from "components";
import { NavigationType } from "interfaces";
import { labels } from "labels";
import { layout } from "styles";

interface Props {
  navigation: NavigationType;
  onPress?: () => void;
  text?: string;
}

const BackButton = (props: Props) => {
  const { text, navigation } = props;
  const { back } = labels.back;

  const onPress = () => navigation.pop();

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress || onPress}
    >
      <Icon name="angle-double-left" style={styles.icon} />
      <Text>{text || back}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    padding: layout.spacing,
  },
  icon: {
    paddingRight: layout.spacingS,
  },
});

export { BackButton };
