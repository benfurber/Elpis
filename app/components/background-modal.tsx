import React, { Component } from "react";
import { ImageBackground, ScrollView, StyleSheet } from "react-native";

import { BackgroundContainer } from "components";
import { colours, layout } from "styles";

const handImagePath = "../assets/images/hand.png";

interface Props {
  children: object;
}

class BackgroundModal extends Component<Props> {
  render() {
    return (
      <BackgroundContainer>
        <ImageBackground
          imageStyle={styles.imageHand}
          source={require(handImagePath)}
          style={styles.containerHand}
        >
          <ScrollView style={styles.container}>
            {this.props.children}
          </ScrollView>
        </ImageBackground>
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colours.whiteTransparentHigh,
    borderTopLeftRadius: layout.borderRadiusL,
    borderTopRightRadius: layout.borderRadiusL,
    width: "100%",
    height: "100%",
    marginTop: layout.spacingL,
  },
  containerHand: {
    width: "100%",
    height: "100%",
    flexDirection: "column-reverse",
    alignItems: "flex-end",
  },
  imageHand: {
    flex: 1,
    width: 360,
    height: 360,
    bottom: 0,
    left: -10,
    opacity: 0.2,
  },
});

export { BackgroundModal };
