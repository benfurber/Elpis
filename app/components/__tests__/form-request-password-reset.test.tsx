import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { FormRequestPasswordReset } from "components";

let navigation;

describe("FormRequestPasswordReset", () => {
  it("renders correctly when empty", () => {
    const component = shallow(
      <FormRequestPasswordReset
        navigation={navigation}
        setRequestedState={() => jest.fn()}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it("renders correctly when email entered", () => {
    const component = shallow(
      <FormRequestPasswordReset
        navigation={navigation}
        setRequestedState={() => jest.fn()}
      />,
    );

    component.setState({ email: "me@me.com" });

    expect(component).toMatchSnapshot();
  });

  it("renders correctly when there's an error", () => {
    const component = shallow(
      <FormRequestPasswordReset
        navigation={navigation}
        setRequestedState={() => jest.fn()}
      />,
    );

    component.setState({ error: { message: "Email not found" } });

    expect(component).toMatchSnapshot();
  });
});
