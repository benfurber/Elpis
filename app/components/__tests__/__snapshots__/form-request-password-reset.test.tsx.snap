// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRequestPasswordReset renders correctly when email entered 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "column",
    }
  }
>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "RequestPasswordReset",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "requestPasswordReset",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "email",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 113,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Mutation>
</View>
`;

exports[`FormRequestPasswordReset renders correctly when empty 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "column",
    }
  }
>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "RequestPasswordReset",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "requestPasswordReset",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "email",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 113,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Mutation>
</View>
`;

exports[`FormRequestPasswordReset renders correctly when there's an error 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "column",
    }
  }
>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "RequestPasswordReset",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "requestPasswordReset",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "email",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 113,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Mutation>
</View>
`;
